sdv/single_table/base.py

```diff
@@ -92,6 +92,70 @@ def __init__(self, metadata, enforce_min_max_values=True, enforce_rounding=True,
         self._fitted_date = None
         self._fitted_sdv_version = None
 
+    def _check_address_initialization(self, anonymization_level):
+        if anonymization_level not in {'full', 'street_address'}:
+            raise ValueError(
+                f"Invalid value '{anonymization_level}' for parameter 'anonymization_level'."
+                " Please provide 'full' or 'street_address'."
+            )
+
+        self._data_processor._check_import_address_transformers()
+
+        if self._fitted:
+            warnings.warn(
+                'Please refit your synthesizer for the address changes to appear in'
+                ' your synthetic data.'
+            )
```

Comment:
I don't think we should raise this until after the address transformers actually get set